woocommerce/woocommerce-ios
# In-app purchases Yosemite implementation

Closes: #7834 


### Description


Implements InAppPurchase store and actions to load the list of available products and purchase a product. Right now, it hardcodes the list of product identifiers and doesn't report any purchases to the API, that will come in #7836 

I attempted to include a list of transactions in the debug view, but I was running into trouble with updates. I'm not sure if it was me not doing concurrency right, or a glitch related to StoreKit testing in Xcode. I reverted those changes for now and might try again if it's useful once we're using the IAP sandbox instead of a local configuration.

### Testing instructions


1. Ensure In-app purchases are enabled in Settings > Experimental features
2. Go to the Hub menu and tap "[Debug] IAP"
3. The debug screen should show a list of products with one product
4. Tapping on that product should bring in the UI to purchase a subscription

To inspect existing transactions, choose "Debug > StoreKit > Manage Transactions" in Xcode to bring up the transaction manager. This is also available from the Debug Area while the app is running:
<img width="444" alt="Screen Shot 2022-10-19 at 12 35 32" src="https://user-images.githubusercontent.com/8739/196668364-e19d35db-5b38-44c7-9962-8e45d2c81b48.png">



### Screenshots



https://user-images.githubusercontent.com/8739/196650757-3e3e8311-6a62-408c-8bfc-895de528f62f.mov



---
